### DISCUSSION TITLE
Question about releasing / publishing

### DISCUSSION THREAD

**Discussion by pdxwebdev** [JanssenProject/jans#8027] [Q&A] [answered]
I have the "happy path" integration completed for Janssen using Agama.

I need next steps to button it up with regards to testing, releasing, and publishing.

This part of the documentation does not appear to be available quite yet. [Make a gama file / Release Project](https://gluu.org/quick-start-guide/#make-agama-file)

Perhaps someone can point me in the right direction. Thanks!

**Comment by nynymike** [comment]
@ossdhaval  ?

**Comment by nynymike** [comment]
@pdxwebdev you should record a loom and share the link :-)

**Comment by ossdhaval** [comment]
@pdxwebdev Is [this](https://docs.jans.io/v1.0.22/admin/developer/agama/quick-start-using-agama-lab/#release-the-project) what you are looking for?

**Reply by pdxwebdev** [reply]
So when I do the release from the interface as instructed in the guide, I get the following error:
![Screenshot from 2024-03-12 11-42-48](https://github.com/JanssenProject/jans/assets/490176/0fc625aa-8249-474a-beb0-b2780e4c21fd)

This is from the dev console in the browser:
```
{
    "code": 422,
    "status": "Unprocessable Entity",
    "message": "Unable to create a release for pre-release."
}
```

This is my project:
https://github.com/pdxwebdev/agama-centeridentity/tree/agama-lab-branch/projects/secretlocationauth

**Comment by nynymike** [comment]
But you can download a `.gama` and successfully deploy it?

**Reply by pdxwebdev** [reply]
Yes, it generates the .Gama file from agama labs interface, and I can import it. That all works. The error happens when I try to do a release or publish from agama labs web interface.

**Reply by nynymike** [reply]
I asked someone from the Agama team to take a look: 

![image](https://github.com/JanssenProject/jans/assets/3717101/18bb6963-635b-405a-84ef-9cd8a869cc7f)

You can check the project contents [here](https://github.com/pdxwebdev/agama-centeridentity/tree/agama-lab-branch/projects/secretlocationauth)

I get the same error when I fork the project and try to release...

One comment here...  did you copy code from the Agama-OpenID project and change the Qualified Name ("QN") of the flows?  You know, if you deploy an Agama project, it is available for you to Trigger from another project. We're working on an example of this in the

**Comment by SafinWasi** [comment, marked as answer]
Hi,
Agama Lab deals with two different types of repos: "release" and "pre-release". The intended workflow is as such:
- You create a blank repo, and initialize it with Agama Lab. This is where you develop flows/upload code/assets. This is the **pre-release repo**. All operations are done on the `agama-lab-branch`. This can contain multiple Agama projects. In your case, the https://github.com/pdxwebdev/agama-centeridentity repo would be a pre-release repo. 
- Create a **different** repo, where you move your work from the pre-release repo via pull requests. This is the **release** repo. Creating releases and publishing can only be done from this type of repo. This repo only contains one project, and everything is stored on the `main` branch.

To release and publish, please create an empty repo with the following properties:
- default branch is named `main`
- repository must be set to public visibility.
- An initial commit is mandatory; the user can achieve this by initializing the repository and adding a README.md file.
- repository must contain a description

Then, open your **pre-release repo** with Agama Lab, and open the project you want to publish. Right click on the file explorer, and select `Publish Project`. Enter the initial version number and the URL of the **release** repo you just created. Agama Lab will then move that project to the release repo and create a release according to the version number.

**Reply by nynymike** [reply]
Makes sense... you can't release from a `pre-release` repo.

**Reply by pdxwebdev** [reply]
Excellent, that worked!
https://github.com/pdxwebdev/agama-secretlocationauth/releases/tag/v0.0.1

Thanks!
